engine/spverrors/unfold.go

```diff
@@ -0,0 +1,92 @@
+package spverrors
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/bitcoin-sv/spv-wallet/models"
+)
+
+type errListUnwrapper interface {
```

Comment:
maybe one of:
- joinedErrors
- wrappedErrorList
- wrappedErrors

Reply:
https://go.dev/doc/effective_go#interface-names

> By convention, one-method interfaces are named by the method name plus an -er suffix or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc.

So, maybe - simple:
- `unwrapper`
- or as I named it
- or inline interface (like they do in `golang` std package:

https://github.com/golang/go/blob/bc047b62b2ec0c6eae3437a26410039c82365f60/src/errors/wrap.go#L62

```golang
if unwrapper, ok := current.(interface { Unwrap() []error }); ok {
```

Reply:
I would distinguish name of the public purpose interfaces from the helper interface that is used for some logic.
Wouldn't it show cleaner intention when you will use something like this:
``
if joined, ok := current.(joinedErrors); ok {
  unfoldJoinedErrors(joined, &result)
  break;
}
```

Reply:
"naming things" 😅 
Let's do it as you with 👌